src/components/Editor/Runners/HtmlRunner/HtmlRunner.js

```diff
@@ -1,84 +1,147 @@
+  );
+  const justLoaded = useSelector((state) => state.editor.justLoaded);
 
-  const dispatch = useDispatch()
+  const dispatch = useDispatch();
   const output = useRef();
+  const [error, setError] = useState(null);
 
+  const showModal = () => {
+    dispatch(showErrorModal());
+  };
 
-  const htmlFiles = projectCode.filter(
-    (component) => component.extension === 'html'
-  )
+  const closeModal = () => setError(null);
 
-  const cssFiles = projectCode.filter(
-    (component) => component.extension === "css"
+  const htmlFiles = projectCode.filter(
+    (component) => component.extension === "html"
   );
 
-  const fileName = openFiles[focussedFileIndex]
-  const focussedComponent = projectCode.find((component) => `${component.name}.${component.extension}` === fileName)
-
+  const fileName = openFiles[focussedFileIndex];
+  const focussedComponent = projectCode.find(
+    (component) => `${component.name}.${component.extension}` === fileName
+  );
 
   const getBlobURL = (code, type) => {
     const blob = new Blob([code], { type });
     return URL.createObjectURL(blob);
   };
 
+  const errorListener = () => {
+    window.addEventListener("message", (event) => {
+      if (typeof event.data === "string" || event.data instanceof String) {
+        if (event.data === "ERROR: External link") {
+          setError("externalLink");
+        }
+      }
+    });
+  };
+
+  useEffect(() => errorListener(), []);
   let timeout;
 
   useEffect(() => {
     if (justLoaded) {
-      runCode()
+      runCode();
     } else {
       timeout = setTimeout(() => {
-        runCode()
+        runCode();
       }, 2000);
-      return () => clearTimeout(timeout)
+      return () => clearTimeout(timeout);
     }
   }, [projectCode, focussedFileIndex]);
 
   useEffect(() => {
     if (codeRunTriggered) {
-      runCode()
+      runCode();
     }
-  }, [codeRunTriggered])
+  }, [codeRunTriggered]);
 
   useEffect(() => {
-    runCode()
-  }, [focussedFileIndex])
+    runCode();
+  }, [focussedFileIndex]);
+
+  useEffect(() => {
+    if (error) {
+      showModal();
+      errorListener();
+    }
+  }, [error]);
 
   const runCode = () => {
     // TODO: get html files and handle urls for non index pages
-    const indexHTML = htmlFiles.find((component) => `${component.name}.${component.extension}` === 'index.html')
-    const componentToPreview = focussedComponent.extension === 'html' ? focussedComponent : indexHTML
-    let indexPage = componentToPreview.content;
-
-    cssFiles.forEach((cssFile) => {
-      const cssFileBlob = getBlobURL(cssFile.content, "text/css");
-      indexPage = indexPage.replace(
-        `href="${cssFile.name}.css"`,
-        `href="${cssFileBlob}"`
+    const indexHTML = htmlFiles.find(
+      (component) => `${component.name}.${component.extension}` === "index.html"
+    );
+    const componentToPreview =
+      focussedComponent.extension === "html" ? focussedComponent : indexHTML;
```

Comment:
I might be missing something here, but is the TODO on line 88 still relevant now? it looks here like we're supporting any focussed file with the `*.html` extension

Reply:
Ah ha! Good spot! That change was merged in from Lois's. I'll remove the TODO 👍